docs/source/deviations-and-extensions.rst

```diff
@@ -0,0 +1,64 @@
+Deviations and Extensions to the OData Spec
+===========================================
+
+There are some minor cases where this library deviates from the official `spec`_, 
+e.g. when the spec is ambiguous, or to add extra non-breaking functionality.
+
+
+Lists with a single item need a trailing comma
+----------------------------------------------
+
+This library needs a trailing comma to represent a list with a single item, 
+whereas the spec does not describe this trailing comma (e.g.
+``(item,)`` instead of ``(item)``). 
+
+The reason is that the spec doesn't seem to differentiate between a list of a
+single item and any other parenthesized expression. This can lead to parsing 
+conflicts. Consider the following expression:
+
+.. code-block::
+	
+	concat(('a'), ('b'))
+
+
+The grammar in the official spec could parse this as either:
+
+* Concatenate 2 lists, the first one containing the single string 'a', the second 
+  one containing the single string 'b'. The result would be ``('a', 'b')``.
+* Concatenate 2 expressions in parentheses. The first one is the string literal
+  'a', the second one the string literal 'b'. The result would be ``'ab'``
+
+
+To make this difference explicit, this library requires a trailing comma to 
+signify a list. The same behavior is present in Python:
+
+>>> ("a")
+'a'
+>>> ("a",)
+('a',)
+
+.. >>> from odata_query.grammar import ODataLexer, ODataParser
+.. >>> lexer = ODataLexer()
+.. >>> parser = ODataParser()
+>>> parser.parse(lexer.tokenize("('a')"))
+String(val='a')
+>>> parser.parse(lexer.tokenize("('a',)"))
+List(val=[String(val='a')])
+
+
+Durations expressed in years and months
+---------------------------------------
+
+The official `spec`_ defines a duration with a number of days, hours, minutes,
+and seconds. E.g. ``duration'P1DT2H'`` is a duration of 1 **D**\ ay and 2 **H**\ ours.
+
+This library adds the ability to express durations containing **Y**\ ears and 
+**M**\ onths. E.g. ``duration'P1Y2M3DT4H'`` would express a duration of 1 **Y**\ ear,
+2 **M**\ onths, 3 **D**\ ays, and 4 **H**\ ours.
+
+It's important to note that the final internal value is still expressed in days,
+based on average durations.
+Thus, a month simply represents 30.44 days, while a year represents 365.25 days.
```

Comment:
Good that this is documented, but I find it unintuitive. What's the practical use of months being 30.44 days and years being 365.25 days?

I would expect things to work like this:

- 2023-01-01 + 1 year = 2024-01-01
- 2024-01-01 + 1 year = 2025-01-01 (even though 2024 is a leap year, so it's 366 days long)

In other words, the duration would only be converted to days when combined with an actual date.
You could argue that this is still ambiguous though.

In Python, `timedelta` does not support months or years for this reason (I guess), but `date.replace` does support changing the year or month.

Reply:
(For context, this was contributed in #53 by someone external. Can't speak for them of course)

I totally agree, I'm guessing this will mainly be used as a shorthand where accuracy doesn't really matter. E.g. "Give me logs of the past 6 months". 
I also agree that the `date.replace` behaviour would probably be more intuitive. But that would also require us to do the date logic in Python and still converting it to a `timedelta` before passing it on to whatever backend the user has chosen, since it's infeasible (or straight up impossible) to express that logic on every backend.